### ISSUE TITLE
Feature request: add eglGetDisplayCallback for allow user can setup display attributes.

### ISSUE THREAD

**Issue by halx99** [glfw/glfw#1718]
*  By default, the google ```angle``` implementation of GLES regards application request using cross-process HWNDs when passing _GLFW_EGL_NATIVE_DISPLAY, but ```Windows 7/Windows Server 2008``` doesn't support the feature without Platform Update (KB2670838).
 please see: https://github.com/glfw/glfw/pull/1717
* If glfw provide a callback, user can setup more display attributes by themself, such as: use eglGetPlatformDisplayEXT to choose dx12 or vulkan as ```angle```'s backend renderer.
 please see: https://github.com/c4games/engine-x/issues/99

**Comment by elmindreda** [comment]
Injecting custom EGL display attributes could be done but seems hairy so I'd like to avoid it if possible.

Instead, basic support for choosing what backend renderer ANGLE uses has been added to (what will be) 3.4 with 0dea8a4441828311af4c31de8ba798ba330c4755.

Is that sufficient for your needs? Do you need other extensions or other options for specific backend renderer extensions?

**Comment by halx99** [comment]
Hi, thanks, I will try it, maybe it's enough

**Comment by elmindreda** [comment]
If there are any specific attributes from those extensions that you also need to set, please let me know. There were too many of them in total for me to want to expose all of them just in case, and since I'm only a sporadic ANGLE user I wouldn't necessarily be able to pick the important ones.

**Comment by halx99** [comment]
thanks, I will try later.

**Comment by halx99** [comment]
Hello, I have test, it's enough for me, but still can't support enable ```EGL_PLATFORM_ANGLE_D3D11ON12_ANGLE```
https://github.com/google/angle/extensions/\EGL_ANGLE_platform_angle_d3d11on12.txt
for some other users.
So, I think you should provide all supported attributes provided by angle renader backend, or provide user to pass the attributes.

**Comment by RustyMoyher** [comment]
I agree. Being able to set `EGL_PLATFORM_ANGLE_D3D11ON12_ANGLE` in GLFW would be very useful.

**Comment by RustyMoyher** [comment]
Hello, following up on this. I still need access to `EGL_PLATFORM_ANGLE_D3D11ON12_ANGLE` attribute. Anything I can do to help? Hoping this makes it’s way into the 3.4 release.

**Comment by combolek** [comment]
FWIW, I summited a PR #2381 that tries to address the D3D11on12 support via a new init hint, without exposing the `EGL_PLATFORM_ANGLE_D3D11ON12_ANGLE` directly.
